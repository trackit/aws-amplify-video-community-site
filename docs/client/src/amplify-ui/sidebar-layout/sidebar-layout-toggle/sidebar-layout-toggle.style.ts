import {MQTablet} from "../../styles/media";
import {css} from "emotion";

export const sidebarLayoutSidebarToggleStyle = css`
  width: 3rem;
  height: 3rem;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  z-index: 10;

  ${MQTablet} {
    display: none;
  }

  .ex-graphic {
    display: none;
  }

  &.in-view {
    .burger-graphic {
      display: none;
    }
    .ex-graphic {
      display: initial;
    }
  }
`;
