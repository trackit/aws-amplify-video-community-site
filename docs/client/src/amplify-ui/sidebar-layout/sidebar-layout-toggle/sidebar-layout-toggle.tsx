import {Component, Host, Prop, h} from "@stencil/core";
import {sidebarLayoutContext} from "../sidebar-layout.context";
import {sidebarLayoutSidebarToggleStyle} from "./sidebar-layout-toggle.style";
import {ToggleInView} from "../sidebar-layout.types";

@Component({tag: "amplify-sidebar-layout-toggle", shadow: false})
export class AmplifySidebarLayoutToggle {
  /*** toggles the state provided by `sidebar-layout` */
  @Prop() readonly toggleInView?: ToggleInView;
  /*** whether or not the sidebar is in view, provided by `sidebar-layout` */
  @Prop() readonly inView?: boolean;
  /*** when the sidebar is in view, what class should be appended to the button? */
  @Prop() readonly inViewClass?: string;

  /**
   * We always render with the class resulting from `sidebarLayoutSidebarToggleStyle`.
   * Depending on whether the sidebar is in view, we want to attach a class
   * to enable conditional styling of the toggle. The toggle user can––for instance––
   * conditionally hide different icons depending on whether they want to open or close the
   * sidebar (mobile only)
   */
  getClass = () => {
    return (
      sidebarLayoutSidebarToggleStyle +
      (this.inViewClass && this.inView ? " " + this.inViewClass : "")
    );
  };

  /**
   * Ensures that we don't trigger the toggle if `this.inView` is truthy.
   * This prevents accidental dual-toggling (which is equivalent to no toggling).
   */
  safeToggle = (e: Event) => {
    if (this.inView) {
      e.stopPropagation();
    }
    if (this.toggleInView) {
      this.toggleInView();
    }
  };

  render() {
    return (
      <Host onClick={this.safeToggle} class={this.getClass()}>
        <slot />
      </Host>
    );
  }
}

sidebarLayoutContext.injectProps(AmplifySidebarLayoutToggle, [
  "inView",
  "toggleInView",
]);
