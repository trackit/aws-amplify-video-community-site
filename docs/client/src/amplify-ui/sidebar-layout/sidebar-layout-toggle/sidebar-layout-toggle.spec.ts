import {AmplifySidebarLayoutToggle} from "./sidebar-layout-toggle";
import {newSpecPage} from "@stencil/core/testing";

describe("amplify-sidebar-layout-toggle", () => {
  describe("Component logic", () => {
    let amplifySidebarLayoutToggle: AmplifySidebarLayoutToggle;
    beforeEach(
      () => (amplifySidebarLayoutToggle = new AmplifySidebarLayoutToggle()),
    );

    it("should init `toggleInView` as `undefined`", () =>
      expect(amplifySidebarLayoutToggle.toggleInView).toBeUndefined());

    it("should init `inView` as `undefined`", () =>
      expect(amplifySidebarLayoutToggle.inView).toBeUndefined());

    it("should init `inViewClass` as `undefined`", () =>
      expect(amplifySidebarLayoutToggle.inViewClass).toBeUndefined());
  });

  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [AmplifySidebarLayoutToggle],
            html: `<amplify-sidebar-layout-toggle />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
