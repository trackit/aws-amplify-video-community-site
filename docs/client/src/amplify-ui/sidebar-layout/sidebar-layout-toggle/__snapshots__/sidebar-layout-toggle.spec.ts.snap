// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-sidebar-layout-toggle Render logic should render 1`] = `
<amplify-sidebar-layout-toggle class="css-y2423j">
  <!---->
</amplify-sidebar-layout-toggle>
`;
